### PR TITLE
nucleo-f334r8: misc spi/gpio improvements

### PR DESCRIPTION
**Pull request by kaspar030**

### CODE REVIEW

**Comment by kaspar030** [comment]
spi untested for now.

File: boards/nucleo-f334/include/periph_conf.h [hunk 1]
```diff
@@ -78,6 +79,69 @@ extern "C" {
...
+#define GPIO_IRQ_15         (-1)    /* not configured */
+
+/* GPIO channel 0 config */
+#define GPIO_0_PORT         GPIOC                   /* Used for user button 1 */
+#define GPIO_0_PIN          13
+#define GPIO_0_CLKEN()      (RCC->AHBENR |= RCC_AHBENR_GPIOCEN)
+#define GPIO_0_EXTI_CFG1()  (SYSCFG->EXTICR[3] &= ~(SYSCFG_EXTICR4_EXTI13))
+#define GPIO_0_EXTI_CFG2()  (SYSCFG->EXTICR[3] |= SYSCFG_EXTICR4_EXTI13_PC)
+#define GPIO_0_IRQ          EXTI15_10_IRQn
+
```
Comment (haukepetersen) [inline boards/nucleo-f334/include/periph_conf.h]: `/** @} */` missing

File: boards/nucleo-f334/include/periph_conf.h [hunk 2]
```diff
@@ -78,6 +79,69 @@ extern "C" {
...
+#define GPIO_IRQ_12         (-1)    /* not configured */
+#define GPIO_IRQ_13         GPIO_0
+#define GPIO_IRQ_14         (-1)    /* not configured */
+#define GPIO_IRQ_15         (-1)    /* not configured */
+
+/* GPIO channel 0 config */
+#define GPIO_0_PORT         GPIOC                   /* Used for user button 1 */
+#define GPIO_0_PIN          13
+#define GPIO_0_CLKEN()      (RCC->AHBENR |= RCC_AHBENR_GPIOCEN)
+#define GPIO_0_EXTI_CFG1()  (SYSCFG->EXTICR[3] &= ~(SYSCFG_EXTICR4_EXTI13))
```
Comment (haukepetersen) [inline boards/nucleo-f334/include/periph_conf.h]: s/`SYSCFG_EXTICR4_EXTI13`/`SYSCFG_EXTICR4_EXTI13_PC`/

**Comment by authmillenon** [comment]
The tests `driver_lis3dh` and `driver_nrf24l01p_lowlevel` are failing to build for `nucleo-f334`:

```
$ BOARD=nucleo-f334 make -C tests/driver_lis3dh/ clean all
make: Entering directory `/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh'
Building application "driver_lis3dh" for "nucleo-f334" with MCU "stm32f3".

/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh/main.c: In function 'main':
<command-line>:0:18: error: 'GPIO_1' undeclared (first use in this function)
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh/main.c:82:23: note: in expansion of macro 'TEST_LIS3DH_INT1'
                       TEST_LIS3DH_INT1, TEST_LIS3DH_INT2, SCALE) == 0) {
                       ^
<command-line>:0:18: note: each undeclared identifier is reported only once for each function it appears in
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh/main.c:82:23: note: in expansion of macro 'TEST_LIS3DH_INT1'
                       TEST_LIS3DH_INT1, TEST_LIS3DH_INT2, SCALE) == 0) {
                       ^
<command-line>:0:18: error: 'GPIO_2' undeclared (first use in this function)
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh/main.c:82:41: note: in expansion of macro 'TEST_LIS3DH_INT2'
                       TEST_LIS3DH_INT1, TEST_LIS3DH_INT2, SCALE) == 0) {
                                         ^
make[1]: *** [/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh/bin/nucleo-f334/driver_lis3dh/main.o] Error 1
make: *** [all] Error 2
make: Leaving directory `/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_lis3dh'
```

```
$ BOARD=nucleo-f334 make -C tests/driver_nrf24l01p_lowlevel/ clean all
make: Entering directory `/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel'
Building application "driver_nrf24l01p_lowlevel" for "nucleo-f334" with MCU "stm32f3".

/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c: In function 'print_register':
<command-line>:0:8: error: 'GPIO_1' undeclared (first use in this function)
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c:100:16: note: in expansion of macro 'CS_PIN'
     gpio_clear(CS_PIN);
                ^
<command-line>:0:8: note: each undeclared identifier is reported only once for each function it appears in
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c:100:16: note: in expansion of macro 'CS_PIN'
     gpio_clear(CS_PIN);
                ^
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c: In function 'cmd_its':
<command-line>:0:8: error: 'GPIO_1' undeclared (first use in this function)
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c:193:56: note: in expansion of macro 'CS_PIN'
     if (nrf24l01p_init(&nrf24l01p_0, SPI_PORT, CE_PIN, CS_PIN, IRQ_PIN) < 0) {
                                                        ^
<command-line>:0:9: error: 'GPIO_2' undeclared (first use in this function)
/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/main.c:193:64: note: in expansion of macro 'IRQ_PIN'
     if (nrf24l01p_init(&nrf24l01p_0, SPI_PORT, CE_PIN, CS_PIN, IRQ_PIN) < 0) {
                                                                ^
make[1]: *** [/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel/bin/nucleo-f334/driver_nrf24l01p_lowlevel/main.o] Error 1
make: *** [all] Error 2
make: Leaving directory `/home/martine/Repositories/RIOT-OS/RIOT/tests/driver_nrf24l01p_lowlevel'
```

**Comment by kaspar030** [comment]
@authmillenon Hm, tests seem to require a certain number of defined GPIOs. Should I blacklist the board? Otherwise, I'll define unused GPIOs, creating code in the gpio implementation for those use cases...

**Comment by kaspar030** [comment]
Sorry for saying SPI was untested before, it was just untested on nucleo-f334. That is working now.

**Comment by authmillenon** [comment]
> @authmillenon Hm, tests seem to require a certain number of defined GPIOs. Should I blacklist the board? Otherwise, I'll define unused GPIOs, creating code in the gpio implementation for those use cases...

If there are more GPIO ports on the board I would prefer solution 2, otherwise if the number of ports are limited: blacklist them ;-)

File: cpu/stm32f3/periph/spi.c [hunk 3]
```diff
@@ -299,20 +309,26 @@ int spi_release(spi_t dev)
...
+    char tmp;
 
-    while (!(spi[dev]->SR & SPI_SR_TXE));
-    spi[dev]->DR = out;
+    /* recast to uint_8 to force 8bit access */
+    volatile uint8_t *DR = (volatile uint8_t*) &spi[dev]->DR;
 
-    while (!(spi[dev]->SR & SPI_SR_RXNE));
+    /* wait for an eventually previous byte to be readily transferred */
+    while(!(spi[dev]->SR & SPI_SR_TXE));
```
Comment (haukepetersen) [inline cpu/stm32f3/periph/spi.c]: actually the RXNE was the safer choice here!

**Comment by haukepetersen** [comment]
Just define more GPIO ports for the board. Even better would be that you create a complete pin mapping for the board, as for the [stm32f4discovery](https://docs.google.com/spreadsheets/d/11iqdux3K1-7FPZChufNgpChwENkgCiLrnKJPoCcJxdg/edit?usp=sharing) or the [stm32f3discovery](https://docs.google.com/spreadsheets/d/1gnxsux5WpFrn-33Ivb9nGgTBqooqgDYxRkhZms-cvsc/edit?usp=sharing)

**Comment by kaspar030** [comment]
-rebased
-repaired stm32f3 isr2 gpio
-added two more GPIOs

File: boards/nucleo-f334/include/periph_conf.h (hunk 1 above)
Comment (LudwigOrtmann) [inline boards/nucleo-f334/include/periph_conf.h]: It is below.

**Comment by haukepetersen** [comment]
ACK when squashed and Travis is happy.

**Comment by kaspar030** [comment]
- rebased
- (tryed to) fix travis static-check complaints

**Comment by kaspar030** [comment]
- squashed

**Comment by kaspar030** [comment]
GO.

**Pull request merged**